src/main/java/com/userManagement/userManagement/mailer/EmailHelper.java

```diff
@@ -0,0 +1,35 @@
+package com.userManagement.userManagement.mailer;
+
+import com.userManagement.userManagement.exception.EmailSendingException;
+import com.userManagement.userManagement.exception.UserManagementException;
+import com.userManagement.userManagement.response.ErrorResponse;
+import jakarta.mail.MessagingException;
+import lombok.SneakyThrows;
+import org.springframework.stereotype.Component;
+
+import java.io.UnsupportedEncodingException;
+
+@Component
+public class EmailHelper {
+    private EmailSender emailSender;
+
+    public EmailHelper(EmailSender emailSender){
+        this.emailSender = emailSender;
+    }
+
+    @SneakyThrows
+    public void sendWelcomeMail(String recipientEmail) {
+        String subject = "Welcome to our application!";
+        String content = "Dear User,\n\nWelcome to our application!\n\nBest regards,\nYour Application Team";
+
+        try {
+            emailSender.sendEmail(recipientEmail, subject, content);
+        } catch (MessagingException e) {
+            ErrorResponse errorResponse = new ErrorResponse("EMAIL_SENDING_ERROR", "Error occurred while sending email.");
```

Comment:
You can directly put object of error respose in EmailSendingException.

Reply:
Create a package enums and inside that one enum class like below and kept all error code and error message

@Getter
public enum ErrorEnum {
    INVALID_USER_ID("100-01","Invalid user id");

    private final String errorCode;
    private final String errorMsg;

    ErrorEnum(final String errorCode, final String errorMsg) {
        this.errorCode = errorCode;
        this.errorMsg = errorMsg;
    }
}